app/actions/remote/reactions.ts

```diff
@@ -30,29 +31,41 @@ export const addReaction = async (serverUrl: string, postId: string, emojiName:
 
     try {
         const currentUserId = await getCurrentUserId(operator.database);
-        const reaction = await client.addReaction(currentUserId, postId, emojiName);
-        const models: Model[] = [];
+        const emojiAlias = getEmojiFirstAlias(emojiName);
+        const reacted = await queryReaction(operator.database, emojiAlias, postId, currentUserId).fetchCount() > 0;
+        if (!reacted) {
+            const reaction = await client.addReaction(currentUserId, postId, emojiAlias);
+            const models: Model[] = [];
+
+            const reactions = await operator.handleReactions({
+                postsReactions: [{
+                    post_id: postId,
+                    reactions: [reaction],
+                }],
+                prepareRecordsOnly: true,
+                skipSync: true, // this prevents the handler from deleting previous reactions
+            });
+            models.push(...reactions);
 
-        const reactions = await operator.handleReactions({
-            postsReactions: [{
-                post_id: postId,
-                reactions: [reaction],
-            }],
-            prepareRecordsOnly: true,
-            skipSync: true, // this prevents the handler from deleting previous reactions
-        });
-        models.push(...reactions);
-
-        const recent = await addRecentReaction(serverUrl, [emojiName], true);
-        if (Array.isArray(recent)) {
-            models.push(...recent);
-        }
+            const recent = await addRecentReaction(serverUrl, [emojiName], true);
+            if (Array.isArray(recent)) {
+                models.push(...recent);
+            }
 
-        if (models.length) {
-            await operator.batchRecords(models);
-        }
+            if (models.length) {
+                await operator.batchRecords(models);
+            }
 
-        return {reaction};
+            return {reaction};
+        }
+        return {
+            reaction: {
+                user_id: currentUserId,
+                post_id: postId,
+                emoji_name: emojiAlias,
+                create_at: Date.now(),
```

Comment:
Probably not a big deal, but if it exist (which is the case) the create_at is not now, is somewhere in the past.

Reply:
true, but is it not used, I'm just adding the value to avoid TS from yelling at me, I can change it to 0